paasta_tools/generate_deployments_for_service.py

```diff
@@ -104,6 +104,31 @@ def get_instance_config_for_service(soa_dir, service):
             )
 
 
+def get_latest_deployment_tag(refs, deploy_group):
+    """Gets the latest deployment tag and sha for the specified deploy_group
+
+    :param refs: A dictionary mapping git refs to shas
+    :param deploy_group: The deployment group to return a deploy tag for
+
+    :returns: A tuple of the form (ref, sha) where ref is the actual deployment
+              tag (with the most recent timestamp)  and sha is the sha it points at
+    """
+    most_recent_dtime = None
+    most_recent_ref = None
+    most_recent_sha = None
+    pattern = re.compile("^refs/tags/paasta-%s-(\d{8}T\d{6})-deploy$" % deploy_group)
+
+    for ref_name, sha in refs.iteritems():
+        match = pattern.match(ref_name)
+        if match:
```

Comment:
ah, I guess we do. this comes from the git repo, so could have any amount of tags/branches. ignore me